utils/openshift_resource.py

```diff
@@ -209,19 +209,19 @@ def initialize_resource_type(self, cluster, namespace, resource_type):
         })
 
     def add_desired(self, cluster, namespace, resource_type, name, value):
-        self._lock.acquire()
-        desired = self._clusters[cluster][namespace][resource_type]['desired']
-        if name in desired:
-            self._lock.release()
-            raise ResourceKeyExistsError(name)
-        desired[name] = value
-        self._lock.release()
+        with self._lock:
+            desired = \
+                self._clusters[cluster][namespace][resource_type]['desired']
+            if name in desired:
+                self._lock.release()
```

Comment:
I wonder if this is necessary. I would assume that once the raise is called, the lock will be automatically released as it exits the `with` context. I may be wrong though...

Reply:
you are right, i just missed it. thanks!